File: Leetcode/1299. Replace Elements with Greatest Element on Right Side/solution.go
func replaceElements(arr []int) []int {
	if len(arr) == 1 {
		result := []int{-1}
		return result
	}

	for i := 0; i < len(arr)-1; i++ {
		arr[i] = max(arr[i+1:])
	}

	arr[len(arr)-1] = -1
	return arr
}

func max(array []int) int {
	var max int = array[0]
	for _, value := range array {
		if max < value {
			max = value
		}
	}
	return max
}

File: Leetcode/1299. Replace Elements with Greatest Element on Right Side/solution.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        if len(arr) == 1:
            return [-1]

        for i in range(len(arr) - 1):
            max_n = max(arr[i + 1:])
            arr[i] = max_n
        arr[-1] = -1
        return arr
